### PR TITLE
Folder name supports LFN

### PR DESCRIPTION
**Pull request by ghost**
My folder is ```D:\Code\Mys```.
The current ```:cd``` also accepts ```D:\code\mys``` and the display is ```D:\code\mys```, this is a folder that does not exist.
The chdir of cmd.exe will be accepted, but will display correctly (```cd C:\code\mys -> C:\Code\Mys>```).

I use GetLongPathName, but it is not supported by OS older than Windows 2000 (Source: MSDN).
That worked well.

I am sorry I let you take precious time. Please consider it.

### CODE REVIEW

**Comment by k-takata** [comment]
We have already dropped support for Windows 2000 or earlier. See v8.0.0029.

**Comment by ghost** [comment]
Difficulty! Overdid it. It will be simpler. Wait a minute.

**Comment by k-takata** [comment]
[The reference](https://docs.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-getlongpathnamew) says:
> It is possible to have access to a file or directory but not have access to some of the parent directories of that file or directory. As a result, GetLongPathName may fail when it is unable to query the parent directory of a path component to determine the long name for that component.

Maybe we should ignore the error of GetLongPathName?

**Comment by ghost** [comment]
Is this OK...?

File: src/os_win32.c [hunk 1]
```diff
@@ -3133,7 +3142,14 @@ mch_dirname(
 	return FAIL;
     }
 #endif
-    return (GetCurrentDirectory(len, (LPSTR)buf) != 0 ? OK : FAIL);
+    if (GetCurrentDirectory(len, (LPSTR)buf) == 0)
+	return FAIL;
+
+    if (GetLongPathNameA((LPSTR)buf, (LPSTR)abuf, _MAX_PATH) == 0)
```
Comment (k-takata) [inline src/os_win32.c:3148]: I think `len` should be used instead of `_MAX_PATH`.

Comment (ghost) [reply src/os_win32.c:3148]: The size of ```abuf``` is ```_MAX_PATH + 1```, so there it is.

Comment (k-takata) [reply src/os_win32.c:3148]: Ah, sorry. You are right.

Comment (ghost) [reply src/os_win32.c:3148]: Thank you so much for everything! :)
I want to improve the quality more.

**Comment by brammool** [comment]
Unfortunately, it looks like Appveyor started failing after this patch was included.
I cannot reproduce it locally.  Also cannot guess what the problem could be.

**Pull request closed without merging**